core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/AbstractHealthCheckedEndpointGroupBuilder.java

```diff
@@ -368,6 +379,52 @@ public SELF selectionTimeoutMillis(long initialSelectionTimeoutMillis, long sele
         return self();
     }
 
+    /**
+     * Sets a predicate to filter health checked {@link Endpoint}s. Whenever there is an update
+     * in endpoints, this predicate is used to filter {@link Endpoint}s that should be considered
+     * for selection.
+     *
+     * <p>For example:<pre>{@code
+     * // regardless of health check status, all endpoints will be considered for selection
+     * HealthCheckedEndpointGroup endpointGroup =
+     *     HealthCheckedEndpointGroup.builder(delegate, "/health")
+     *                               .healthCheckedEndpointPredicate(endpoint -> true)
+     *                               .build();
+     * }</pre>
+     */
+    @UnstableApi
+    public SELF healthCheckedEndpointPredicate(Predicate<Endpoint> healthCheckedEndpointPredicate) {
```

Comment:
> If this API is used in xDS now, how about adding it after https://github.com/line/armeria/pull/5741 is merged?

I didn't think #5741 is related to this PR since I didn't think `DEGRADED` is not really used internally at the moment. I thought the more important benefit of this PR was that `UNHEALTHY` endpoints are now considered for priority calculation.

> Because I think the health status needs to be exposed explicitly to users to implement this healthCheckedEndpointPredicate.

I see. Am I understanding correctly that you prefer this API is not exposed publicly unless `HealthCheckedEndpoint` is introduced?

Reply:
> I thought the more important benefit of this PR was that UNHEALTHY endpoints are now considered for priority calculation.

I looked over quicky your PR. Now I understood your purpose. 

> you prefer this API is not exposed publicly unless HealthCheckedEndpoint is introduced?

I thought at least an utility API should be provided to check the health status of an `Endpoint` if it takes some time to design `HealthCheckedEndpoint`.

It seems like no one will be using this API right now, so it is okay to add it. But changes for `HealthCheckedEndpoint` would be needed in a f/u work to make this API useful to normal users.